File: backend/src/main/java/by/fpmibsu/findafriend/controller/commands/shelters/UpdateShelterHandler.java
package by.fpmibsu.findafriend.controller.commands.shelters;

import by.fpmibsu.findafriend.application.mediatr.RequestHandler;
import by.fpmibsu.findafriend.dataaccesslayer.shelter.ShelterDao;
import by.fpmibsu.findafriend.entity.Place;

public class UpdateShelterHandler extends RequestHandler<Boolean, UpdateShelterCommand> {
    private final ShelterDao shelterDao;

    public UpdateShelterHandler(ShelterDao shelterDao) {
        this.shelterDao = shelterDao;
    }

    @Override
    public Boolean handle(UpdateShelterCommand request) {
        var shelter = shelterDao.getEntityById(request.shelterId);
        if (shelter == null) {
            return false;
        }
        if (request.name != null) {
            shelter.setName(request.name);
        }
        if (request.placeId != -1) {
            var p = new Place();
            p.setId(request.placeId);
            shelter.setPlace(p);
        }
        if (request.website != null) {
            shelter.setWebsite(request.website);
        }
        if (request.address != null) {
            shelter.setAddress(request.address);
        }

        shelterDao.update(shelter);
        return true;
    }
}

File: BackendNET/Backend/Commands/Shelters/UpdateShelter/UpdateShelterHandler.cs
using Backend.Repository;
using MediatR;

namespace Backend.Commands.Shelters.UpdateShelter;

public class UpdateShelterHandler : IRequestHandler<UpdateShelterCommand, bool>
{
    private readonly IShelterRepository _shelterRepository;

    public UpdateShelterHandler(IShelterRepository shelterRepository)
    {
        _shelterRepository = shelterRepository;
    }

    public async Task<bool> Handle(UpdateShelterCommand request, CancellationToken cancellationToken)
    {
        var instance = await _shelterRepository.GetByIdAsync(request.ShelterId, cancellationToken);
        if (instance is null)
        {
            return false;
        }

        instance.Name = request.Name ?? instance.Name;
        instance.Address = request.Address ?? instance.Address;
        instance.Website = request.Website ?? instance.Website;
        instance.PlaceId = request.PlaceId ?? instance.PlaceId;
        return true;
    }
}